### PR TITLE
[1.10] DCOS-37555: prevent unnecessary updates in ServiceBreadcrumb

### PR DESCRIPTION
**Pull request by Poltergeist**
<details><summary><strong>fix: prevent service breadcrumb to update without changes</strong></summary><hr/>

This adds a `shouldComponentUpdate` function to the `ServiceBreadcrumb` component. This ensures the
component is only updated if any of the props has changed.

To test you can use the following service config:

```JSON
{
  "id": "/volumes",
  "containers": [
    {
      "name": "container-1",
      "resources": {
        "cpus": 0.1,
        "mem": 128,
        "disk": 0
      },
      "image": {
        "kind": "DOCKER",
        "id": "alpine"
      },
      "exec": {
        "command": {
          "shell": "sleep 1000;"
        }
      },
      "volumeMounts": [
        {
          "name": "home",
          "mountPath": "/var/home"
        },
        {
          "name": "nothome",
          "mountPath": "/var/nothome"
        }
      ]
    }
  ],
  "volumes": [
    {
      "name": "home",
      "persistent": {
        "type": "root",
        "size": 1,
        "constraints": []
      }
    },
    {
      "name": "nothome",
      "persistent": {
        "type": "root",
        "size": 1,
        "constraints": []
      }
    }
  ],
  "networks": [
    {
      "mode": "host"
    }
  ],
  "scaling": {
    "instances": 1,
    "kind": "fixed"
  },
  "scheduling": {
    "placement": {
      "constraints": []
    }
  },
  "executorResources": {
    "cpus": 0.1,
    "mem": 32,
    "disk": 10
  },
  "fetch": []
}
```

This is going to create a multi container application with multiple volumes. Then you should
navigate to the service detail page.
Check if you can navigate to the task detail page and back.
Check if you can visit the volume detail page.
Check if you can switch between the two volume detail pages.
  - Copy the URL of the first volume Detail page,
  - Visit the second one, go to the address bar paste the URL and hit enter.
  - After this you may use the browser history to go back and forth.

Close DCOS-22453
</details>

---
Commit:
87fa23929e0cf158cafb10dd1c36867720bee13c

### CODE REVIEW

**Comment by mesosphere-mergebot-dev** [comment]
This repo has @mesosphere-mergebot integration. You can interact with the following commands.
```
@mesosphere-mergebot integrate  
@mesosphere-mergebot ship-it  
@mesosphere-mergebot test  
```

**Comment by Poltergeist** [comment]
Backport of #2962

File: plugins/services/src/js/components/ServiceBreadcrumbs.js [hunk 1]
```diff
@@ -43,6 +42,22 @@ class ServiceBreadcrumbs extends React.Component {
     global.addEventListener("resize", this.handleViewportResize);
   }
 
+  shouldComponentUpdate(nextProps) {
+    const hasServiceIDChanged = this.props.serviceID !== nextProps.serviceID;
```
Comment (weblancaster) [inline plugins/services/src/js/components/ServiceBreadcrumbs.js]: same suggestion as https://github.com/dcos/dcos-ui/pull/2962#discussion_r189669460

File: plugins/services/src/js/components/ServiceBreadcrumbs.js [hunk 2]
```diff
@@ -43,6 +42,22 @@ class ServiceBreadcrumbs extends React.Component {
...
+    const hasTaskIDChanged = this.props.taskID !== nextProps.taskID;
+    const hasTaskNameChanged = this.props.taskName !== nextProps.taskName;
+    const hasExtraNotChanged =
```
Comment (weblancaster) [inline plugins/services/src/js/components/ServiceBreadcrumbs.js]: same suggestion as https://github.com/dcos/dcos-ui/pull/2962#discussion_r189669460

**Comment by juliangieseke** [comment]
hold merge until master got merged 👍

**Comment by DanielMSchmidt** [comment]
@juliangieseke I strongly disagree with blocking this until master is merged. 1.10 is our top priority, so it should be the other way around.

**Comment by juliangieseke** [comment]
@DanielMSchmidt if this is true I'd ask why this is a Backport, it should be a fix. opening backport PRs for stuff that isnt merged yet only creates confusion (like now) :(

**Comment by DanielMSchmidt** [comment]
@juliangieseke I agree, we should only have had one PR in the first place, have it merged and do forward ports afterwards

**Comment by Poltergeist** [comment]
Closing this one in favor of #2962 since the discussion is happening there.

**Comment by DanielMSchmidt** [comment]
@Poltergeist We should have the discussion here, 1.10 is our top priority right now, AFAIK

**Comment by Poltergeist** [comment]
Since now the master PR #2962 got merged I am reopening this PR.

**Pull request merged**